projects/ngx-skeleton-loader/src/lib/ngx-skeleton-loader.component.ts

```diff
@@ -24,24 +27,41 @@ export class NgxSkeletonLoaderComponent implements OnInit, AfterViewInit, OnDest
   static ngAcceptInputType_animation: boolean | string;
 
   @Input()
-  count = 1;
+  count: NgxSkeletonLoaderConfig['count'];
 
   @Input()
-  loadingText = 'Loading...';
+  loadingText: NgxSkeletonLoaderConfig['loadingText'];
 
   @Input()
-  appearance: 'circle' | '' = '';
+  appearance: NgxSkeletonLoaderConfig['appearance'];
 
   @Input()
-  animation: 'progress' | 'progress-dark' | 'pulse' | 'false' | false = 'progress';
+  animation: NgxSkeletonLoaderConfig['animation'];
 
-  // This is required since ngStyle is using `any` as well
-  // More details in https://angular.io/api/common/NgStyle
-  // tslint:disable-next-line: no-any
-  @Input() theme: { [k: string]: any } = {};
+  @Input()
+  theme: NgxSkeletonLoaderConfigTheme;
 
   // tslint:disable-next-line: no-any
-  items: Array<any> = [];
+  items: Array<any>;
+
+  private config: NgxSkeletonLoaderConfig;
+
+  constructor(@Inject(NGX_SKELETON_LOADER_CONFIG) @Optional() config: NgxSkeletonLoaderConfig) {
+    this.config = {
+      appearance: 'line',
+      animation: 'progress',
+      theme: {},
+      loadingText: 'Loading...',
+      count: 1,
+      ...config
+    };
+    this.count = this.config.count;
+    this.loadingText = this.config.loadingText;
+    this.appearance = this.config.appearance;
+    this.animation = this.config.animation;
+    this.theme = this.config.theme ?? null;
```

Comment:
If you allow me to give you a suggestion, you can simplify this to:

```suggestion
    this.count = config?.count ?? 1;
    this.loadingText = config?.loadingText ?? 'Loading...';
    this.appearance = config?.appearance ?? 'line';
    this.animation = config?.animation ?? 'progress';
    this.theme = config?.theme ?? null;
```